sdk/src/driftClient.ts

```diff
@@ -446,11 +451,19 @@ export class DriftClient {
 
 		// Update provider for txSender with new wallet details
 		this.txSender.provider = newProvider;
-
 		this.wallet = newWallet;
 		this.provider = newProvider;
 		this.program = newProgram;
 		this.authority = newWallet.publicKey;
+		this.activeSubAccountId = activeSubAccountId;
+		this.userStatsAccountPublicKey = undefined;
+
+		this.users.clear();
+		await this.addAllUsers(
```

Comment:
why are you adding the users before the unsubscribe? feel like this is backwards?

Reply:
mistake, not sure how it wasn't breaking anything